foundation-models/openai/src/test/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClientTest.java

```diff
@@ -445,17 +498,26 @@ void streamChatCompletionDeltas() throws IOException {
       // Configure the HttpClient mock to return the mock response
       doReturn(mockResponse).when(httpClient).executeOpen(any(), any(), any());
 
-      final var request =
-          new OpenAiChatCompletionParameters()
-              .addMessages(
-                  new OpenAiChatUserMessage()
-                      .addText("Can you give me the first 100 numbers of the Fibonacci sequence?"));
+      final var userMessage =
+          new ChatCompletionRequestUserMessage()
+              .role(ChatCompletionRequestUserMessage.RoleEnum.USER)
+              .content(
+                  ChatCompletionRequestUserMessageContent.create(
+                      "Can you give me the first 100 numbers of the Fibonacci sequence?"));
+      final var request = new CreateChatCompletionRequest().addMessagesItem(userMessage);
 
       try (Stream<OpenAiChatCompletionDelta> stream = client.streamChatCompletionDeltas(request)) {
 
-        OpenAiChatCompletionOutput totalOutput = new OpenAiChatCompletionOutput();
-        final List<OpenAiChatCompletionDelta> deltaList =
-            stream.peek(totalOutput::addDelta).toList();
```

Comment:
(Major/Comment)

I found the `totalOutput` handling confusing. I did not write any basic convenience for that, instead I made sure the information can be extracted. With casting and/or custom-field lookups. This needs refinement.